src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMReadyToRunGenericHelperNode.cs

```diff
@@ -122,17 +112,17 @@ protected sealed override void EmitCode(NodeFactory factory, ref ARMEmitter enco
                         {
                             // We need to trigger the cctor before returning the base. It is stored at the beginning of the non-GC statics region.
                             GenericLookupResult nonGcRegionLookup = factory.GenericLookup.TypeNonGCStaticBase(target);
-                            EmitDictionaryLookup(factory, ref encoder, encoder.TargetRegister.Arg0, encoder.TargetRegister.Arg0, nonGcRegionLookup, relocsOnly);
+                            EmitDictionaryLookup(factory, ref encoder, encoder.TargetRegister.Arg1, encoder.TargetRegister.Arg2, nonGcRegionLookup, relocsOnly);
 
                             int cctorContextSize = NonGCStaticsNode.GetClassConstructorContextStorageSize(factory.Target, target);
-                            encoder.EmitSUB(encoder.TargetRegister.Arg0, cctorContextSize);
-                            // cmp [r0 + ptrSize], 1
-                            encoder.EmitLDR(encoder.TargetRegister.Arg2, encoder.TargetRegister.Arg0, factory.Target.PointerSize);
-                            encoder.EmitCMP(encoder.TargetRegister.Arg2, 1);
+                            encoder.EmitSUB(encoder.TargetRegister.Arg2, cctorContextSize);
+                            // cmp [r2 + ptrSize], 1
+                            encoder.EmitLDR(encoder.TargetRegister.Arg3, encoder.TargetRegister.Arg2, factory.Target.PointerSize);
```

Comment:
Can this be `encoder.EmitLDR(encoder.TargetRegister.Arg3, encoder.TargetRegister.Arg0, factory.Target.PointerSize - cctorContextSize);` instead?

Reply:
That's true, I'll fix, many thanks